pandas/core/algorithms.py

```diff
@@ -548,6 +567,9 @@ def mode(values):
 
     dtype = values.dtype
     if is_integer_dtype(values):
+        # This also works if dtype is uint64 because there is a 1-1
+        # correspondence between int64 and uint64, and we will cast
+        # all elements back to their correct uint64 values.
         values = _ensure_int64(values)
```

Comment:
I would rather make this less magical and just select based on dtype. A read will be confused by this; better to do it on dtype.